docs/4.4/api-reference.md

```diff
@@ -113,50 +113,200 @@ func main() {
 
 ## Roles
 
-[Roles](enterprise/ssh-rbac.md#roles) are used to define what resources and actions a user is allowed/denied access to. It's best to make your role's permissions as stringent as possible.
+Every user in Teleport is assigned a set of [roles](enterprise/ssh-rbac.md#roles). A user's roles defines what actions or resources the user is allowed or denied to access.
 
-**Retrieve role**
+Some of the permissions a role could define include:
+
+* Which SSH nodes a user can or cannot access.
+* Ability to replay recorded sessions.
+* Ability to update cluster configuration.
+* Which UNIX logins a user is allowed to use when logging into servers.
+
+!!! Note
+    The open source edition of Teleport automatically assigns every user to the built-in `admin` role but the Teleport Enterprise allows administrators to define their own roles with far greater control over the user permissions.
+
+You can manage roles with the Teleport CLI tool  [tctl](cli-docs.md#tctl), or programmatically with the RPC calls documented below.
+
+You may want to use the API to manage roles if:
+
+- You want to write a program that can always ensure certain roles exist on your system and do not want to orchestrate tctl to do this.
+- You want to dynamically create short lived roles.
+- You want to dynamically create roles with fields filled that Teleport currently does not support.
+
+### The Role Object
+
+Before we break down the Role object piece by piece, You can look at the `Create Role` section below to see an example of a role.
+
+A Teleport `role` is primarily defined by its `Spec`, which contains its `Allow` rules, `Deny` rules, and OpenSSH `Options`.
 
 ```go
-role, err := client.GetRole("auditor")
+// RoleV3 represents role resource specification
+type RoleV3 struct {
+  // Spec is the role specification
+  Spec RoleSpecV3 struct {
+    // Options is for OpenSSH options like agent forwarding.
+    Options RoleOptions
+    // Allow is the set of conditions evaluated to grant access.
+    Allow RoleConditions
+    // Deny is the set of conditions evaluated to deny access. Deny takes priority over allow.
```

Comment:
It's probably worth adding a note here (or in the "further explanation" section below) that if a user has multiple roles, the `Deny` section applies to the full set of permissions that they get as a result.

This catches people out sometimes; they set up one role for dev access which explicitly denies `production` and a second role that explicitly allows `production`, but they still don't get the `production` access they expect when they have both roles (because of the explicit `Deny` on the first role which applies to everything)